README.md

```diff
@@ -7,6 +7,85 @@ Version 2.0. See the file "[LICENSE](LICENSE)" for more information.
 
 ## Introduction
 
+mod-settings is a key-value store. It is an attempt to
+make a more secure configuration system than what is
+available today in FOLIO.
+
+See [Fixing the security problem in mod-configuration](https://github.com/MikeTaylor/folio-docs/blob/main/doc/fixing-mod-configuration.md)
+for more information.
+
+It is currently implemented with PostgresQL as storage.
+
+A setting consists of these required properties
+
+* id: unique identifier (UUID)
+* scope: like namespace for the setting. This could be module.
+* key: a readable identifier; not necessarily for display
+* value: a JSON object
+
+And optionally:
+
+* userId: the owner of the setting
+
+We call settings without userId 'global'. global settings must be unique
+for scope and key. Non-global settings must be unique for scope, key
+and userId.
+
+Settings are protected by permissions.
+
+In order to write to a global setting, the client must have permission
+'settings.global.write.'scope, where scope what is used in the setting.
+To read this type of setting, the client must have
+'settings.global.read.'scope.
+
+In order to write to a setting with any userId, which we call a user
+setting, the client must have permission 'settings.users.write.'scope ,
+where scope what is used in the setting. To read this type of setting,
+the client must have 'settings.users.read.'scope.
+
+In order to write to user's own setting, the client must have permission
+'settings.own.write.'scope, where scope what is used in the setting.
+To read this type of setting, the client must have 'settings.own.read.'scope.
+
+With 'read' in this contexts we mean able to read the content of
+the setting. With 'write' in this context we modify storage.
+
+The API is CRUD-like, but with some important changes for some.
+
+Create a setting with:
+
+    POST /config/entries
+
+This is write operation and returns 204 if successful.
+
+Fetch a particular setting with:
+
+    GET /config/entries/{id}
+
+This fetch is protected by permissions. If the client does not
+have permission to read the setting with the scope, it will return
+a 4XX failure.
```

Comment:
Can we say _which_ 4XX?

Reply:
Yeah.. I know.. The "problem" is speed. If we want things to be fast, it may return 404 in cases where, perhaps, it should be 403. Example: DELETE setting. It will be (not know) limited by rights.. SO .. WHERE ID=id AND (scope = XX OR scope = YY).. The result is going to be that it's "not found". When in reality it is there. Question is whether we want this to be a two times SQL query or one. See? Should I say 403 or 404 , for now?

Reply:
Putting 403 / 404 for  now.

Reply:
What you have done makes sense. On the other hand, I imagine attempts to set permissions will be very uncommon, rare enough that the overhead of another DB access to determine the correct error code will surely be negligible. So we could either go ahead and do that, so that we always report the correct error; or, as you say, just say "403 or 404".

Reply:
If you want full security it not even be revealed.. If you get 403 you then know that identifier is out there, so perhaps it cane revealed somehow. If you get 404 you know it's not there (for any permission).

Reply:
Ah! That is a good argument for _always_ returning 404.

(I've seen this done in other places. It feels wrong at first, but makes sense. If we do that, we ought to mention in the docs that we're doing it for this security reason.)

Reply:
ok. Will update